grotain.go

```diff
@@ -10,38 +21,32 @@ type Cards struct {
 
 // CalcInterval calculates the next revision in days
 func (c Cards) CalcInterval(quality int) int {
-
-	if quality < 3 {
-		return 1
-		// return int(math.Round(float64((c.interval / 2))))
+	if quality < wrongResponse {
+		return newIntervalAfterWrongAnswer
 	}
 
-	// for n>2 I(n):=I(n-1)*EF
 	switch c.Interval {
-	case 0:
-		return 1
-	case 1:
-		return 6 // 3?
+	case firstInterval:
+		return newFirstInterval
+	case secondInterval:
+		return newSecondInterval
 	default:
-		return int(math.Round(float64((c.Interval - 1)) * c.Efactor))
+		return int(math.Round(float64((c.Interval - oneDayAgo)) * c.Efactor))
 	}
 }
 
 // CalcEfactor calculates the new easiness factor reflecting the easiness of memorizing and retaining a given item in memory
 func (c Cards) CalcEfactor(quality int) float64 {
-	// EF':=EF+(0.1-(5-q)*(0.08+(5-q)*0.02))
 	c.Efactor = c.Efactor + (0.1 - float64(5-quality)*(0.08+float64(5-quality)*float64(0.02)))
```

Comment:
Aqui tem mais um que dá pra refatorar, não?

Reply:
Ainda não sei como posso refatorar essa linha, pois esse é o algoritmo padrão chamado [SM-2](https://www.supermemo.com/en/archives1990-2015/english/ol/sm2), que é representado por esse cálculo `EF':=EF+(0.1-(5-q)*(0.08+(5-q)*0.02))`. Talvez, o que posso fazer nesse momento seria isso:

```suggestion
	c.Efactor = algorithmSm2(c.Efactor, quality)
```

e adicionar uma fun para isso:

```golang
func algorithmSm2(efactor float64, quality int) float64 {
	return efactor + (0.1 - float64(5-quality)*(0.08+float64(5-quality)*float64(0.02)))
}
```

Nos próximos PRs penso em configurar uma interface para deixar o método `CalcEfactor` mais flexível com outros possíveis algoritmos, mas não gostaria de fazer isso exatamente nesse PR para não misturar as coisas.

Bom, mas gostaria de ouvir sua opinião:

1. Deixo por enquanto assim, sem mudar nada? Já que logo vou configurar a possibilidade de utilizar outros algoritmos.
2. Extrair para uma `func algorithmSm2`?
3. outra ideia? ☺️ 

Obrigado @masisiliani

Reply:
A solução que você propôs já uma boa! Nomear números/ calculos e encapsular em funções é uma boa prática de refact ;)

Fazendo essa alteração pra mim tá :shipit: :)